addons/mail/static/src/new/core/chat_window_model.js

```diff
@@ -35,6 +70,12 @@ export class ChatWindow {
             _state: state,
         });
         this.update(data);
+        if (ChatWindow.maxVisibleChatWindows(this._state) <= this._state.chatWindows.length) {
+            const visibleChatWindows = ChatWindow.visibleChatWindows(this._state);
+            const swaped = visibleChatWindows[visibleChatWindows.length - 1];
+            swaped.hidden = true;
+            swaped.folded = true;
```

Comment:
strange to sync folded with hidden.

Also other methods that do show/hide call fold methods which also update fold state on thread, whereas from here the thread is not updated according to change in chat window.